cmd/kube-proxy/app/options/options.go

```diff
@@ -1,105 +0,0 @@
-/*
-Copyright 2014 The Kubernetes Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-// Package options contains flags for initializing a proxy.
-package options
-
-import (
-	_ "net/http/pprof"
-	"time"
-
-	utilfeature "k8s.io/apiserver/pkg/util/feature"
-	clientv1 "k8s.io/client-go/pkg/api/v1"
-	"k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/apis/componentconfig"
-	"k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1"
-	_ "k8s.io/kubernetes/pkg/features"
-	"k8s.io/kubernetes/pkg/kubelet/qos"
-	"k8s.io/kubernetes/pkg/util"
-
-	// add the kubernetes feature gates
-	_ "k8s.io/kubernetes/pkg/features"
-
-	"github.com/spf13/pflag"
-)
-
-// ProxyServerConfig configures and runs a Kubernetes proxy server
-type ProxyServerConfig struct {
-	componentconfig.KubeProxyConfiguration
-	ResourceContainer string
-	ContentType       string
-	KubeAPIQPS        float32
-	KubeAPIBurst      int32
-	ConfigSyncPeriod  time.Duration
-	CleanupAndExit    bool
-	NodeRef           *clientv1.ObjectReference
-	Master            string
-	Kubeconfig        string
-}
-
-func NewProxyConfig() *ProxyServerConfig {
-	versioned := &v1alpha1.KubeProxyConfiguration{}
-	api.Scheme.Default(versioned)
-	cfg := componentconfig.KubeProxyConfiguration{}
-	api.Scheme.Convert(versioned, &cfg, nil)
-	return &ProxyServerConfig{
-		KubeProxyConfiguration: cfg,
-		ContentType:            "application/vnd.kubernetes.protobuf",
-		KubeAPIQPS:             5.0,
-		KubeAPIBurst:           10,
-		ConfigSyncPeriod:       15 * time.Minute,
-	}
-}
-
-// AddFlags adds flags for a specific ProxyServer to the specified FlagSet
-func (s *ProxyServerConfig) AddFlags(fs *pflag.FlagSet) {
-	fs.Var(componentconfig.IPVar{Val: &s.BindAddress}, "bind-address", "The IP address for the proxy server to serve on (set to 0.0.0.0 for all interfaces)")
```

Comment:
> Most flags have been removed and replaced by a single config file

I agree with @thockin that we need to preserve the flag behavior accroding to our deprecation policy so we can't remove options.go just yet.

Can you do the same thing here but bind to sub fields?